### PR TITLE
Migration: Gitlab: Support Subdirectory

### LINKED ISSUE #13535: Wrong URL when importing from GitLab hosted in subdirectory
Gitea v1.13.0-rc2

Note that in the `GET` request `gitlab/` is missing after the domain name.

(not sure if the error message is related to this though)

```
Migrating from http://my.gitlab.domain/gitlab/user/test.git failed.

GET https://my.gitlab.domain/api/v4/projects/gitlab/user/test: 502 failed to parse unknown error format
```

### PR DESCRIPTION
**Pull request by 6543**
close #13535

### CODE REVIEW

File: modules/migrations/gitlab.go [hunk 1]
```diff
@@ -87,6 +88,23 @@ func NewGitlabDownloader(ctx context.Context, baseURL, repoPath, username, passw
...
+	// split namespace and subdirectory
+	pathParts := strings.Split(strings.Trim(repoPath, "/"), "/")
+	for len(pathParts) > 2 {
+		if _, _, err = gitlabClient.Version.GetVersion(); err == nil {
+			break
+		}
+
+		baseURL = path.Join(baseURL, pathParts[0])
+		pathParts = pathParts[1:]
+		err = gitlab.WithBaseURL(baseURL)(gitlabClient)
```
Comment (6543) [inline modules/migrations/gitlab.go]:
```suggestion
		_ = gitlab.WithBaseURL(baseURL)(gitlabClient)
```

**Review by lafriks — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Although I think this might conflict with any attempt at namespace support.

We might be better off forcing the user to tell us how to spilt up these URLs

**Comment by 6543** [comment]
@zeripath it should not :) 

In short it will test the baseUrl until an api responces if not it add the first repo-path part to the base url

Since this testing do not care if it is a repo with namespaces or not we should be fine

**Comment by 6543** [comment]
Backport: #13591

**Pull request merged**